Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/fragments/GameSettingsFragment.java

```diff
@@ -0,0 +1,99 @@
+package org.dolphinemu.dolphinemu.fragments;
+
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.os.Environment;
+import android.preference.ListPreference;
+import android.preference.PreferenceFragment;
+
+import org.dolphinemu.dolphinemu.NativeLibrary;
+import org.dolphinemu.dolphinemu.R;
+import org.dolphinemu.dolphinemu.activities.GameDetailsActivity;
+import org.dolphinemu.dolphinemu.utils.Log;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStreamWriter;
+
+
+public class GameSettingsFragment extends PreferenceFragment implements SharedPreferences.OnSharedPreferenceChangeListener {
+	/**
+	 * The fragment argument representing the section number for this
+	 * fragment.
+	 */
+	private static final String ARG_GAME_ID = "game_id";
+	private static final String ARG_SECTION_NUMBER = "section_number";
+
+	private GameDetailsActivity mActivity;
+
+
+	public GameSettingsFragment() {
+	}
+
+	/**
+	 * Returns a new instance of this fragment for the given section
+	 * number.
+	 */
+	public static GameSettingsFragment newInstance(int sectionNumber, String gameID) {
+		GameSettingsFragment fragment = new GameSettingsFragment();
+		Bundle args = new Bundle();
+		args.putInt(ARG_SECTION_NUMBER, sectionNumber);
+		args.putString(ARG_GAME_ID, gameID);
+		fragment.setArguments(args);
+		return fragment;
+	}
+
+	private void loadPrefFromIni(String prefKey, String sect, String key)
+	{
+		ListPreference targetLp = (ListPreference) findPreference(prefKey);
+
+		targetLp.setValue(NativeLibrary.GetUserSetting(getArguments().getString("game_id"), sect, key));
+	}
+
+	private void savePrefToIni(String prefKey, String sect, String key)
+	{
+		ListPreference targetLp = (ListPreference) findPreference(prefKey);
+
+		NativeLibrary.SetUserSetting(getArguments().getString("game_id"), sect, key, targetLp.getValue());
+	}
+
+	@Override
+	public void onCreate(Bundle savedInstanceState)
+	{
+		super.onCreate(savedInstanceState);
+
+		addPreferencesFromResource(R.xml.preferences);
+		loadPrefFromIni("list_preference_enable_dual_core", "Core", "CPUThread");
+		loadPrefFromIni("list_preference_enable_mmu", "Core", "MMU");
+		loadPrefFromIni("list_preference_skip_dbcz", "Core", "DBCZ");
+		loadPrefFromIni("list_preference_enable_mmu", "Core", "FPRF");
+		loadPrefFromIni("list_preference_sync_gpu", "Core", "SyncGPU");
+		loadPrefFromIni("list_preference_speedup_dtr", "Core", "FastDiscSpeed");
+		loadPrefFromIni("list_preference_dsp_hle", "Core", "DSPHLE");
```

Comment:
This is a kinda weird set of settings to show in a per-game config dialog. I know that DolphinWX uses the same set, but it lacks many useful settings and has some settings that you likely never will need to change...

@JMC47, what do you think? Just try to copy DolphinWX for now, or take the opportunity to think things through while we're making a new GUI for it anyway?

Reply:
I was simply copying DolphinWX for a starting point, but yeah, more relevant settings would be desirable.